sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_models.py

```diff
@@ -223,12 +227,19 @@ def __init__(self, **kwargs):
 
     @classmethod
     def _from_generated(cls, entity):
+        offset = entity.offset
+        length = entity.length
+        if isinstance(entity, _v3_0_models.Entity):
+            # we do not return offset and length for v3.0 since
+            # the correct encoding was not introduced for v3.0
+            offset = None
```

Comment:
is this the pattern other libraries have taken? I am new to this paradigm as in TA and FR we can only support one version of the SDK

Reply:
for python, we're supposed to support all API versions, the user only needs to pass in the api_version as a kwarg if they don't want the default.